nobody community-impact-proposal

PeterDaveHello opened the pull request.

joenash commented:
Thanks Peter! 

> I'm not going to make the increasing the community size a main goal

Great to hear, I think you've got a really clear vision of the priorities important to you, which is awesome. 

One thing I would be cautious of, though, is the age/year group of your currently engaged students. One place you may still want to make a focused effort to grow is with the younger year groups, because older members graduate and leave the club. 

> We start from collecting the super easy to solve [issues] (https://github.com/cdnjs/cdnjs/issues/9248) for the students

Your approach here is so good! One of the biggest problems for participants in Hacktoberfest and other events, is finding accessible issues. I love that you're making sure all of your members have a good first experience. 

When they go ahead and make these changes and this pull request, are they doing that alone or with you at the workshop session?

> Some helpful swags(not sure if I should call it swag), like GitHub Git cheat sheet poster and stickers, I already received some from GitHub Education.

Let me know if you need more!

> Foods and drinks for workshop and hackathon, I could buy some.

This is also something we can help with :) 

## Summary

I think this is a really solid plan, Peter! You've got all the parts in place in this proposal, covering the content, timeline, format, online spaces and support from TAs. I think the core challenge for you will be the **why**: once a student wants to commit to open source, your plan is a great introduction. But how do you take them from not knowing about open source, to wanting to commit? Though as you mentioned at the beginning, it sounds like you have enough students finding that out on their own, so right now, I think this proposal is 👍 good to go! 👍

PeterDaveHello commented:
Thanks Joe! 

> One thing I would be cautious of, though, is the age/year group of your currently engaged students. One place you may still want to make a focused effort to grow is with the younger year groups, because older members graduate and leave the club. 

I have the same concern on this part and still trying to find a way to deal with this complex issue, as this group is different with the groups of speeches and talks, we more focus on leading the new students to real contributions, some how it'll take more time and pressure then just to listen to a speeches, but the good news is, we now have 1st grade students in the team, so they'll still have about 3~5 years in the school :) Even the graduated students, I think the current new students, after one or two years, they can still invite us graduated students to give a talk or lesson, non regular, with much more experience :D 

> When they go ahead and make these changes and this pull request, are they doing that alone or with you at the workshop session?

As you can see in that issue (though the checklist there is now empty), the students will usually take 2~4 issues there, we'll make sure they can follow up and at least finish a pull request, if there are problems, we'll help and make sure the problem been solved in the class, this can assure we have a good start for everybody, prove that sending an acceptable PR is not that hard. (The other issues will be their "homework" )

> Let me know if you need more!

No problem, since I received lots of stickers and the cheat sheets, I think it's enough right now :D

> This is also something we can help with :) 

It'll be very cool, we ca discuss this part in Slack later!

> I think the core challenge for you will be the **why**: once a student wants to commit to open source, your plan is a great introduction. But how do you take them from not knowing about open source, to wanting to commit? Though as you mentioned at the beginning, it sounds like you have enough students finding that out on their own, so right now, I think this proposal is 👍 good to go! 👍 

Actually, we introduced about the concept of free and open source software in the beginning, and also introduced the most famous free and open source projects just around us, in the real life, like Mozilla Firefox, Chromium, Git, Linux kernel, AOSP, blah ... the impact is not only on the GitHub profile, but also the career life, and the real collaboration in the projects can really help we learn the experience and techniques to help us be much more efficient and strong, the school works will also benefits from this experience.

I am wondering if I should invite some famous FOSS contributors to give us talks to inspire them, actually I invited @muan and @johndbritton to give us a talk in Yuan Ze University about two years ago, I believe that they did inspire the students, even not only the students but also the teachers, the problem is, to invite those famous guys, we usually think not only the small group can learn from them, so the event could be bigger, than we'll need to prepare to advertise the events, try to get a large classroom, talk to the department, the paper works loading will be much more higher, that's a trade off problem, I'll try to figure it out :)

Thanks for your review in detail again!

muan commented:
I'm now based in Taipei and would be more than happy to take part if the meetup happens in Taipei. Feel free to let me know if I can be of any help. :)
